table/block_based/block_based_table_builder.cc

```diff
@@ -340,58 +337,57 @@ struct BlockBasedTableBuilder::Rep {
   uint64_t get_offset() { return offset.load(std::memory_order_relaxed); }
   void set_offset(uint64_t o) { offset.store(o, std::memory_order_relaxed); }
 
-  const IOStatus& GetIOStatus() {
-    if (compression_opts.parallel_threads > 1) {
-      std::lock_guard<std::mutex> lock(status_mutex);
-      return io_status;
+  bool IsParallelCompressionEnabled() const {
+    return compression_opts.parallel_threads > 1;
+  }
+
+  Status GetStatus() {
+    if (status_ok.load(std::memory_order_relaxed)) {
```

Comment:
Why using `memory_order_relaxed` here and also all the other places? Maybe add comments for that.

Reply:
Added comments. It's due to both write and read of status are protected by status_mutex, so no strict memory order is required to make the status change visible before status_ok changes.

Reply:
sorry, just to confirm, do you mean `status_ok` doesn't have to be strong consistent between threads?
Seems write to `status_ok` is protected by mutex, but not read like here.

Reply:
Yes.

`status_ok` indicates the state of `status`, so we only needs to keep modifications before `status_ok` change visible. However, those modifications will always be visible because they're protected by a mutex. So we don't need to require the memory order of `status_ok`.

Reply:
Got it, maybe we should mention it the comments that `we need to keep modifications of status before status_ok`.

Reply:
Sure, updated.

Reply:
Updated.